server/base/src/main/java/org/apache/accumulo/server/metrics/MetricsInfoImpl.java

```diff
@@ -180,98 +100,75 @@ public void addMetricsProducers(MetricsProducer... producer) {
           "called addMetricsProducers() without providing at least one producer - this has no effect");
       return;
     }
-    lock.lock();
-    try {
-      if (composite == null) {
-        producers.addAll(Arrays.asList(producer));
-      } else {
-        Arrays.stream(producer).forEach(p -> p.registerMetrics(composite));
-      }
-    } finally {
-      lock.unlock();
-    }
-  }
 
-  @Override
-  public MeterRegistry getRegistry() {
-    lock.lock();
-    try {
-      if (composite == null) {
-        throw new IllegalStateException("metrics have not been initialized, call init() first");
-      }
-    } finally {
-      lock.unlock();
+    if (commonTags == null) {
+      producers.addAll(Arrays.asList(producer));
+    } else {
+      Arrays.stream(producer).forEach(p -> p.registerMetrics(Metrics.globalRegistry));
     }
-    return composite;
   }
 
   @Override
-  public void init() {
+  public synchronized void init(Collection<Tag> tags) {
```

Comment:
It looks like `addMetricsProducers` is almost always called before `init`. I wonder if we should just change `init` to `init(Collection<Tag> tags, MetricsProducer... producers)` and get rid of `addMetricProducers`.

Reply:
I'll give that a try, if it works it would be a nice improvement.

Reply:
Looked into it and there is once place that `addMetricsProducer` is called that is not close to the init code.  That place is in the thrift code where some thrift metrics are added.